test/uk/gov/hmrc/cataloguefrontend/createappconfigs/CreateAppConfigsControllerSpec.scala

```diff
@@ -182,7 +180,69 @@ class CreateAppConfigsControllerSpec
 
       status(result) shouldBe 303
       redirectLocation(result) shouldBe Some(routes.ServiceCommissioningStatusController.getCommissioningState(serviceName).url)
+    }
+
+    "return 303 and redirect when service requires mongo but slug info is not found" in new Setup {
 
+      when(authStubBehaviour.stubAuth(any[Option[Predicate.Permission]], eqTo(Retrieval.EmptyRetrieval)))
+        .thenReturn(Future.successful(true))
+
+      when(mockTRConnector.repositoryDetails(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Some(gitRepository)))
+
+      when(mockSCSConnector.commissioningStatus(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Some(List.empty[Check])))
+
+      when(mockSDConnector.getSlugInfo(any[String], any[Option[Version]])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(None))
+
+      when(mockGHConnector.getGitHubProxyRaw(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Some("mongo")))
+
+      when(mockBDConnector.createAppConfigs(form.copy(appConfigBase = true), serviceName, ServiceType.Backend, requiresMongo = true, isApi = false))
+        .thenReturn(Future.successful(Right(AsyncRequestId("requestId"))))
+
+      val result = controller
+        .createAppConfigs(serviceName)(
+          FakeRequest(POST, "/create-app-configs")
+            .withSession(SessionKeys.authToken -> "Token token")
+            .withFormUrlEncodedBody("appConfigBase" -> "true")
+        )
+
+      status(result) shouldBe 303
+      redirectLocation(result) shouldBe Some(routes.ServiceCommissioningStatusController.getCommissioningState(serviceName).url)
+    }
+
+    "return 303 and redirect when service does not require mongo" in new Setup {
+
+      when(authStubBehaviour.stubAuth(any[Option[Predicate.Permission]], eqTo(Retrieval.EmptyRetrieval)))
+        .thenReturn(Future.successful(true))
+
+      when(mockTRConnector.repositoryDetails(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Some(gitRepository)))
+
+      when(mockSCSConnector.commissioningStatus(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Some(List.empty[Check])))
+
+      when(mockSDConnector.getSlugInfo(any[String], any[Option[Version]])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(None))
+
+      when(mockGHConnector.getGitHubProxyRaw(any[String])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Some("")))
```

Comment:
it will be clearer what is going on if you return `Some(dependenciesWithoutMongo)` where `dependenciesWithoutMongo = ""`

Reply:
Theres a lot of duplication with the mongo case, consider something like
```scala
  "redirect" when {
    List(true, false).foreach { withMongo =>
      (if (withMongo) "with mongo" else "without mongo") in {
        ...
        when(mockGHConnector.getGitHubProxyRaw(any[String])(any[HeaderCarrier]))
           .thenReturn(Future.successful(Some(if (withMongo) "" else "mongo"))
        ...
        verify(mockBDConnector)..createAppConfigs(requiresMongo = withMongo)
   ```